backend/src/utils/groupsUtils.ts

```diff
@@ -0,0 +1,88 @@
+import { CustomObjectsApi } from '@kubernetes/client-node';
+import { setDashboardConfig } from '../routes/api/config/configUtils';
+import {
+  GroupCustomObject,
+  groupObjResponse,
+  GroupsConfigBody,
+  KubeFastifyInstance,
+} from '../types';
+import { getDashboardConfig, updateDashboardConfig } from './resourceUtils';
+
+export class MissingGroupError extends Error {
+  constructor(message: string) {
+    super(message);
+    Object.setPrototypeOf(this, MissingGroupError.prototype);
+  }
+}
+
+export const getGroupsCR = (): GroupsConfigBody => {
+  try {
+    return getDashboardConfig().spec.groupsConfig;
+  } catch (e) {
+    throw new Error(`Failed to retrieve Dashboard CR groups configuration`);
+  }
+};
+
+export const updateGroupsCR = async (
+  fastify: KubeFastifyInstance,
+  groupsConfigBody: GroupsConfigBody,
+): Promise<GroupsConfigBody> => {
+  try {
+    await setDashboardConfig(fastify, {
+      spec: {
+        dashboardConfig: getDashboardConfig().spec.dashboardConfig,
+        groupsConfig: groupsConfigBody,
+      },
+    });
+    return getGroupsCR();
```

Comment:
mmm weird, i remember i changed it cause if I returned the object returned in `setDashboardConfig` it wasn't updated, but it seems I mischecked it or something cause I'm checking it now and it seems to be working fine.
But I will argue that is important to always return the latest persited object instead of just not updating the client, it solves several issues with race conditions or multiple updates to the same values.